tex/generic/pgf/systemlayer/pgfsys-common-postscript.def

```diff
@@ -231,7 +202,7 @@
           \hss%
         \noexpand\pgfsys@endpurepicture}}%
   }%
-  \global\expandafter\let\csname @pgfshading#1!\endcsname=\pgfutil@tempa
+  \global\expandafter\let\csname @pgfshading#1!\endcsname=\pgfutil@tempa%
```

Comment:
No trailing `%` required, same for the other similar lines below.

Reply:
There are many other assignments (not by me, `master` branch) that are terminated with `%`, e. g.

````latex
% Shadings

\def\pgfsys@horishading#1#2#3{%
  {%
    \pgf@parsefunc{#3}%
    \pgfmathsetlength\pgf@x{#2}%
    \pgf@xa=\pgf@x%                        <---------------
    \pgf@sys@bp@correct{\pgf@x}%
    \pgf@y=\pgf@max%                       <---------------
````

Reply:
In this case it actually does no harm, but it's bad style.  I think it's better to know the rules when TeX will skip spaces and when it continues scanning.